.pre-commit-config.yaml

```
@@ -44,6 +44,12 @@ repos:
         # pre-commit's default_language_version, see
         # https://pre-commit.com/#top_level-default_language_version
         language_version: python3.10
+  - repo: https://github.com/python-poetry/poetry
+    rev: '1.4.0'
+    hooks:
+      - id: poetry-check
+      - id: poetry-lock
+        language_version: python3.10
```

Add this if you don't want a frustrating experience with this pre-commit hook
```suggestion
        language_version: python3.10
        args: ['--check']
```

---

done